urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/yuccasrv/Yucca.java

```diff
@@ -0,0 +1,1099 @@
+	
+    }
+    
+    public void performUnlisten( BasicType socketID ) 
+	throws BasicTypeException,
+	       IOException {
+
+	//System.out.println( "Converting "+socketID.getClass().getName()+" to UUID: "+socketID );
+	UUID uuid = socketID.getUUID();
+	this.bindManager.release( uuid ); 
+	
+    }
+
+
+    public void run() {
+
+	try {
+	    
+	    System.out.println( "Starting command line ..." );
+	    // The method will block here
+	    this.yuccaLine.runCommandLine();
+
+	
+	} catch( IOException e ) {
+
+	    e.printStackTrace();
+	}
+
+
+    }
+
+
+    public static String processCustomizedFilePath( String filePath ) {
+	if( filePath == null )
+	    return null;
+
+	filePath = filePath.replaceAll( "\\{USER_HOME\\}", System.getProperty("user.home") );
+	// ...
+	
+	return filePath;
+    }
+	
+
+    private static boolean performFirstRun( File configurationFile ) 
+	throws IOException {
+
+	File configurationParentFile = configurationFile.getParentFile();
+
+
+	System.out.println( "It seems this is the first time you are running yucca." );
+	System.out.println( "Yucca needs to create a configuration directory at '" + configurationFile.getPath() + "' and store some config files inside." );
+	System.out.print( "Is that okay? " );
+
+	BufferedReader in = new BufferedReader( new InputStreamReader(System.in) );
+	String line = null;
+	do {
+	    
+	    System.out.println( "Type 'y' or 'n': " );
+	    line = in.readLine();
+
+	} while( !line.equals("y") && !line.equals("n") );
+
+	
+	if( line.equals("n") )
+	    return false;
+
+
+	File configurationTemplateDirectory = new File( "_.templates" );
+	boolean filesCopied = FileCopy.copy( configurationTemplateDirectory, configurationParentFile );
+	
+		
+	// Configuration directory successfully created?
+	return filesCopied;
+    }
+
+    private static void printSingleLineBox( String[] lines ) {
+
+	// Find longest line
+	String line = null;
+	for( int i = 0; i < lines.length; i++ ) {
+	    if( line == null || (line != null && lines[i] != null && lines[i].length() > line.length()) )
+		line = lines[i];
+	}
+
+	if( line == null )
+	    return; // No lines at all?!
+
+	// Characters in Unicode
+	// Does work :)
+	StringBuffer b = new StringBuffer();
+	b.append( '\u250C' );  // left upper corner
+	for( int i = 0; i < line.length(); i++ )
+	    b.append( '\u2500' );  // single horizonal line
+	b.append( '\u2510' );  // right upper corner
+	b.append( "\n" );
+	for( int l = 0; l < lines.length; l++ ) {
+
+	    b.append( '\u2502' );  // single vertical line
+	    b.append( lines[l] );
+	    for( int k = lines[l].length(); k < line.length(); k++ )
+		b.append( " " );
+	    b.append( '\u2502' );
+	    b.append( "\n" );
+	}
+
+	b.append( '\u2514' );  // left lower corner
+	for( int i = 0; i < line.length(); i++ )
+	    b.append( '\u2500' );  
+	b.append( '\u2518' );  // right lower corner
+	//b.append( "\n" );
+	
+	System.out.println( b.toString() );
+    }
+
+    public static Yucca runYucca( String[] argv ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 24. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 1019-1054, lines 1057-1070 into separate methods.